addons/website_forum/views/website_forum.xml

```diff
@@ -1426,7 +1426,7 @@
 
     <div t-attf-class="o_wforum_author_box d-inline-flex #{display_info and 'o_show_info'} #{compact and 'o_compact align-items-center'} #{bio_popover_data and 'o_wforum_bio_popover'}"
          t-att-data-content="bio_popover_data">
-        <t t-set="user_profile_url" t-valuef="#"/>
+        <t t-set="user_profile_url" t-value="'/forum/%s/user/%s' % (slug(forum), object.create_uid.id) + '?forum_origin=' + request.httprequest.path"/>
```

Comment:
That's not correct:
1. Indeed the current behavior might looks odd, but it's not really a bug, there is no traceback nor blocking flow. So this should not be changed in stable.
2. That would probably not be the best way to fix it in master either, we would probably prefer to transform the `<a/>` to something else like a `<span>` (to check) to remove the link, simply. So there wouldn't be any "odd behavior when clicking on that link" since there wouldn't be any link anymore. That would probably be better than landing on a private / not readable page.
3. When you read the code, that's actually the behavior that is wanted: "if it is not published, don't set the url" 
4. This looks weird when reading it:
```
a = "something"
if condition:
    a = "something"
```